### PR TITLE
22211 - add amalgamation spec

### PR DESCRIPTION
**Pull request by ketaki-deodhar**
*Issue #:* /bcgov/entity#22211

*Description of changes:*
Add amalgamation (regular and short form) API spec endpoints

image

image

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: docs/business.yaml
Comment (ketaki-deodhar) [inline docs/business.yaml:1810]: Based on the new structure, added Registration example under `\businesses` path. Paul's ticket will add IA

File: docs/business.yaml
Comment (ketaki-deodhar) [inline docs/business.yaml:1811]: Based on the new structure, added Registration example under `\businesses` path. Paul's ticket will add IA

File: docs/business.yaml
Comment (eason-pan-bc) [inline docs/business.yaml:1738]: same as above, consider a mocked id

File: docs/business.yaml
Comment (eason-pan-bc) [inline docs/business.yaml:1755]: typo

File: docs/business.yaml
Comment (eason-pan-bc) [inline docs/business.yaml:1769]: Based on the response in the Postman collection, seems like missing `courtApproval`

File: docs/business.yaml
Comment (eason-pan-bc) [inline docs/business.yaml:1794]: same as above, consider using a mocked id

File: docs/business.yaml
Comment (eason-pan-bc) [inline docs/business.yaml:1811]: typo.

Also, I couldn't find the registration request in the Postman collection. Would you mind to add it there just to keep them consistent?

File: docs/business.yaml
Comment (eason-pan-bc) [inline docs/business.yaml:1891]: Consider adding these 2 failed examples to the Postman collection?

File: docs/business.yaml
Comment (eason-pan-bc) [inline docs/business.yaml:1994]: Based on the request body in the Postman collection. Seems like missing a `deliveryAddress` after or before the `mailingAddress`.

File: docs/business.yaml
Comment (eason-pan-bc) [inline docs/business.yaml:2081]: Based on the request body in the Postman collection, missing the `courtApproval` here

File: docs/business.yaml
Comment (eason-pan-bc) [inline docs/business.yaml:2102]: type mismatches this request `amalgamation-short-form-vertical-request`

File: docs/business.yaml
Comment (eason-pan-bc) [inline docs/business.yaml:1891]: Would you mind to add these 2 failed examples to the Postman collection? May help users to understand different situations.

File: docs/business.yaml
Comment (eason-pan-bc) [inline docs/business.yaml:1994]: Based on the request body in the Postman collection, missing a `deliveryAddress` here

File: docs/business.yaml
Comment (eason-pan-bc) [inline docs/business.yaml:2029]: please fix the indentation for `amalgamationApplication`

File: docs/business.yaml
Comment (eason-pan-bc) [inline docs/business.yaml:2081]: Based on the request body in the Postman collection, missing `courtApproval` after `contactPoint`

File: docs/business.yaml
Comment (eason-pan-bc) [inline docs/business.yaml:2102]: Mismatched type here, suppose to be `vertical`

File: docs/business.yaml
Comment (eason-pan-bc) [inline docs/business.yaml:2550]: Consider adding these items to the amalgamation schema? Since they are in the Business Schemas. ( https://github.com/bcgov/business-schemas/blob/main/src/registry_schemas/schemas/amalgamation_application.json#L108C1-L121C30 )

File: docs/business.yaml
Comment (ketaki-deodhar) [reply docs/business.yaml:1811]: This is not new. It was there before but I had to move it under here as we have changed the structure how we add examples

File: docs/business.yaml
Comment (ketaki-deodhar) [reply docs/business.yaml:1663]: deliveryAddress is added above mailingAddress

File: docs/business.yaml
Comment (ketaki-deodhar) [reply docs/business.yaml:1778]: Short form horizontal response does not include` courtApproval`

File: docs/business.yaml
Comment (ketaki-deodhar) [reply docs/business.yaml:1638]: see below

File: docs/business.yaml
Comment (ketaki-deodhar) [reply docs/business.yaml:1902]: When a user is Not authorized to Registration or amalgamation for given NR or account Id. I can add `not-authorized-to-submit` to the name
image

File: docs/business.yaml
Comment (ketaki-deodhar) [reply docs/business.yaml:2564]: added

File: docs/business.yaml
Comment (eason-pan-bc) [reply docs/business.yaml:1902]: or maybe mention the root cause is due to use an accountID or  NR that is not authorized to the user? I was just thinking about how to make it easier to understand.

File: docs/business.yaml
Comment (eason-pan-bc) [reply docs/business.yaml:2564]: not sure, do we also need this part in the new added `offices`? 
image

File: docs/business.yaml
Comment (ketaki-deodhar) [reply docs/business.yaml:1902]: I think this explains the reason well enough

File: docs/business.yaml
Comment (ketaki-deodhar) [inline docs/business.yaml:1904]: Re: above

File: docs/business.yaml
Comment (ketaki-deodhar) [reply docs/business.yaml:2564]: I think it's not always required only for Regular

File: docs/business.yaml
Comment (ketaki-deodhar) [reply docs/business.yaml:1902]: > Would you mind to add these 2 failed examples to the Postman collection? May help users to understand different situations.

Added Fail examples to Postman for each type

**Review by eason-pan-bc — APPROVED** [review]
LGTM 😃 , but it's better to wait for vysakh to take a final review before merge.

File: docs/business.yaml
Comment (ketaki-deodhar) [reply docs/business.yaml:2564]: added

File: docs/business.yaml
Comment (vysakh-menon-aot) [inline docs/business.yaml:1909]: `type` is not required.

**Review by vysakh-menon-aot — CHANGES_REQUESTED** [review]

File: docs/business.yaml
Comment (ketaki-deodhar) [reply docs/business.yaml]: removed all

**Review by vysakh-menon-aot — APPROVED** [review]

**Pull request merged**
